packages/http/src/mocker/index.ts

```diff
@@ -41,20 +51,52 @@ const mock: IPrismComponents<IHttpOperation, IHttpRequest, IHttpResponse, IMockH
 
       return config;
     }),
-    chain(mockConfig => negotiateResponse(mockConfig, input, resource)),
-    chain(result => assembleResponse(result, payloadGenerator))
+    Reader.chain(mockConfig => negotiateResponse(mockConfig, input, resource)),
+    Reader.chain(result => assembleResponse(result, payloadGenerator)),
+    Reader.chain(response =>
+      withLogger(logger =>
+        pipe(
+          response,
+          Either.map(r => runCallbacks({ resource, request: input.data, response: r })(logger)),
+          Either.chain(() => response)
```

Comment:
Ok this is becoming difficult to review because of types error that are happening because of missing packages (updated types and http spec) — but this is likely wrong. 

I think in this case you need to use `ReaderEither` and one of its internal mapping methods to avoid doing all this.

### Welcome to monad composition :)